appinventor/components-ios/src/extensions/posenet.swift

```diff
@@ -0,0 +1,255 @@
+  fileprivate var _initialized: Bool = false
+  fileprivate var _enabled: Bool = true
+  fileprivate var _backgroundImage: String = ""
+  
+  private enum IllegalStateError: Error {
+    case webviewerNotSet
+  }
+  
+  
+  public override init(_ parent: ComponentContainer) {
+    super.init(parent)
+    _keyPoints["nose"] = []
+    _keyPoints["leftEye"] = []
+    _keyPoints["rightEye"] = []
+    _keyPoints["leftEar"] = []
+    _keyPoints["rightEar"] = []
+    _keyPoints["leftShoulder"] = []
+    _keyPoints["rightShoulder"] = []
+    _keyPoints["leftElbow"] = []
+    _keyPoints["rightElbow"] = []
+    _keyPoints["leftWrist"] = []
+    _keyPoints["rightWrist"] = []
+    _keyPoints["leftHip"] = []
+    _keyPoints["rightHip"] = []
+    _keyPoints["leftKnee"] = []
+    _keyPoints["rightKnee"] = []
+    _keyPoints["leftAnkle"] = []
+    _keyPoints["rightAnkle"] = []
+  }
+  
+  /**
+   Configure the current webview. Add a WKScriptMessageHandler
+   to handle the following Javascript calls necessary for
+   Posenet Extension:
+          ready()
+          reportImage(String dataURL)
+          reportResult(String result)
+          error(int errorCode, String errorMessage)
+   Equivalent to the @Javascript interface in the Android Posenet Extension.
+   */
+  private func configureWebView(_ webview: WKWebView) {
+    self._webview = webview
+    let config = webview.configuration
+    // config.defaultWebpagePreferences.allowsContentJavaScript = true
+    let controller = config.userContentController
+    controller.add(self, name: "ready")
+    controller.add(self, name: "reportImage")
+    controller.add(self, name: "reportResult")
+    controller.add(self, name: "error")
+  }
+  
+  
+  // MARK: PoseNet Properties
+  
+  
+//  @objc open var WebViewer: WebViewer {
+//    get {
+//      return _webviewer
+//    }
+//    set {
+//      if 
+//    }
+//  }
+//  
+  
+  
+  
+  
+  @objc open var MinPoseConfidence: Double {
+    get {
+      return _minPoseConfidence
+    }
+    set {
+      _minPoseConfidence = newValue
+      if _initialized {
+        do {
+          try assertWebView("MinPoseConfidence")
+        } catch {
+          print(_ERROR_WEBVIEWER_NOT_SET, "MinPoseConfidence")
+        }
+      }
+    }
+  }
+  
+  @objc open var MinPartConfidence: Double {
+    get {
+      return _minPartConfidence
+    }
+    set {
+      _minPartConfidence = newValue
+      if _initialized {
+        do {
+          try assertWebView("MinPartConfidence")
+        } catch {
+          print(_ERROR_WEBVIEWER_NOT_SET, "MinPartConfidence")
+        }
+      }
+    }
+  }
+  
+  @objc open var KeyPoints: [String] {
+    get {
+      var keyPoints: [[String]] = []
+      for point in _keyPoints.values {
+        if point.count == 2 {
+          keyPoints.append(point)
+        }
+      }
+      // TODO: build yaillist?
+      // HELP: what is YailList.makelist returning?
+      // returning flat array?
+      return keyPoints.flatMap { $0 }
```

Comment:
Generally, for iOS you shouldn't have to construct a YailList directly. If you return a Swift Array that should be sufficient and the SchemeKit library should handle the interoperability.